source/pod.go

```diff
@@ -154,7 +154,28 @@ func (ps *podSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, error
 		if ps.podSourceDomain != "" {
 			domain := pod.Name + "." + ps.podSourceDomain
 			if len(targets) == 0 {
-				addToEndpointMap(endpointMap, domain, suitableType(pod.Status.PodIP), pod.Status.PodIP)
+				ept := EndpointsTypePodIP
```

Comment:
Any chance to cover all the lines with unit tests?

![Screenshot 2025-05-10 at 08 27 52](https://github.com/user-attachments/assets/1c205efd-5729-4532-a935-9a4213c76864)